src/ucs/sys/sys.c

```diff
@@ -488,38 +487,77 @@ void ucs_get_mem_page_size(void *address, size_t size, size_t *min_page_size_p,
             continue;
         }
 
-        if (start > (uintptr_t)address + size) {
-            /* the scanned range is after memory range of interest - stop */
-            break;
-        }
-        if (end <= (uintptr_t)address) {
-            /* the scanned range is still before the memory range of interest */
-            continue;
+        if (address != NULL) {
+            if (start > (uintptr_t)address + size) {
+                /* the scanned range is after memory range of interest - stop */
+                break;
+            }
+            if (end <= (uintptr_t)address) {
+                /* the scanned range is still before the memory range of interest */
+                continue;
+            }
         }
 
+        info.start = start;
+        info.end   = end;
+
         while (fgets(buf, sizeof(buf), file) != NULL) {
             n = sscanf(buf, "KernelPageSize: %lu kB", &page_size_kb);
-            if (n < 1) {
-                continue;
+            if (n == 1) {
+                info.page_size = page_size_kb * UCS_KBYTE;
             }
 
-            page_size = page_size_kb * UCS_KBYTE;
-            if (found) {
-                *min_page_size_p = ucs_min(*min_page_size_p, page_size);
-                *max_page_size_p = ucs_max(*max_page_size_p, page_size);
-            } else {
-                found            = 1;
-                *min_page_size_p = page_size;
-                *max_page_size_p = page_size;
+            n = 9;
+            if (memcmp(buf, "VmFlags: ", n) == 0) {
+                while (buf[n] != '\n' && n < sizeof(buf)) {
+                    if (memcmp(buf + n, "dc", 2) == 0) {
+                        info.flags |= UCS_MEM_FLAG_DONTCOPY;
+                    }
```

Comment:
can we parse the flags with strtok? to be able to support more flags in the future